Fix `save_meshio` with Numpy v2

keurfonluu opened the pull request:
### Overview

- Fixed: `np.split` returns a tuple of arrays in v2 instead of a list (see discussion #5533).

MatthewFlamm reviewed (approved):
This change keeps the status quo, so it is fine.  I still suspect some sort of error or inefficiency here, but I am not well aware of the meshio format to offer detailed suggestions.

keurfonluu commented:
> This change keeps the status quo, so it is fine. I still suspect some sort of ~error or~ inefficiency here, but I am not well aware of the meshio format to offer detailed suggestions.

I replaced `np.split` with a simple comprehension list. Looks faster.

```python
import numpy as np

def split(x, indices):
    return [x[i1 : i2] for i1, i2 in zip(np.insert(indices, 0, 0), np.append(indices, len(x)))]

np.random.seed(0)
n = 10000
x = np.random.rand(n)
indices = np.sort(np.random.choice(n, size=n // 100, replace=False))

xn = list(np.split(x, indices))
xc = split(x, indices)

assert len(xn) == len(xc)
for x1, x2 in zip(xn, xc):
    assert np.allclose(x1, x2)

%timeit list(np.split(x, indices))
# 91.4 µs ± 6.81 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit split(x, indices)
# 43.6 µs ± 2.45 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

MatthewFlamm reviewed (approved):
Quite a minor speedup, but thank you for looking into it.  I think I was making a mountain over a molehill.
